### PR TITLE
WT-3113 Feature: add a way to dump the cache at runtime.

### PR DESCRIPTION
**Pull request by keithbostic**
@agorrod, @michaelcahill, here's a first cut at a framework that would allow us to dump the cache in running systems.

It looks like statistics logging: there's a directory, and diagnostic commands write into a file in that directory. The file is named `WiredTigerDiag.XX.day.hour`, where XX is the diagnostic command, so you'll see something like: `WiredTigerDiag.cache_dump.05.18`.

Michael, I drilled a hole through`WT_CONNECTION.reconfigure` as you suggested. I added two new configuration strings: `diagnostic` which is a list of commands (the only currently supported one is `cache_dump`), and `diagnostic_path` which allows you to declare a place to store the diagnostic output.

Let me know what you think.

### CODE REVIEW

**Comment by michaelcahill** [comment]
I've put a comment on the ticket: https://jira.mongodb.org/browse/WT-3113.

**Comment by keithbostic** [comment]
@agorrod, I think this branch is ready for review, and I've updated [the discussion in the ticket](https://jira.mongodb.org/browse/WT-3113?focusedCommentId=1476380&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-1476380).

Review notes:
* I removed the new `WT_SESSION.reconfigure(diagnostic)` command, and replaced it with two new verbose configurations, `dump_cache` and `dump_txn`.
* I'm dumping the transaction/cache information every 5 minutes, as long as we're stuck.
* I added locking around the cache  dump's walk of the connection's handle list, but I didn't do anything fancy (like increment the handle's inuse counter and release the lock), I just locked down the list.
* The dumped output  is via `__wt_msg()` calls, which resolve to the informational message handler (`stdout` in the case of `test/format`). I think that works in all cases, for example, gdb, but it is a change from the previous behavior which used `stderr`.
* I don't think any locking is required to walk the transaction list, so I didn't add any.

**Comment by agorrod** [comment]
This switches some code in the eviction server from diagnostic only. I've moved our Jenkins perf tests onto this branch, let's watch the output there.

@keithbostic Have you tested this change out through MongoDB? I'd like to see a sample of what the output looks like for a MongoDB user before pulling the trigger on merging.

**Comment by keithbostic** [comment]
> Have you tested this change out through MongoDB?

Here's what the output looks like in the MongoDB log:
```
./mongo --eval "db.adminCommand({setParameter:1, wiredTigerEngineRuntimeConfig : 'verbose=(dump_txn)'})"
./mongo --eval "db.adminCommand({setParameter:1, wiredTigerEngineRuntimeConfig : 'verbose=(dump_cache)'})"



2017-01-17T15:17:45.506+0000 I NETWORK  [thread1] connection accepted from 127.0.0.1:45110 #14 (1 connection now open)
2017-01-17T15:17:45.506+0000 I NETWORK  [conn14] received client metadata from 127.0.0.1:45110 conn14: { application: { name: "MongoDB Shell" }, driver: { name: "MongoDB Internal Client", version: "3.5.1-154-g166a966" }, os: { type: "Linux", name: "Amazon Linux AMI release 2016.03", architecture: "x86_64", version: "Kernel 4.4.15-25.57.amzn1.x86_64" } }
2017-01-17T15:17:45.506+0000 I COMMAND  [conn14] command admin.$cmd appName: "MongoDB Shell" command: isMaster { isMaster: 1, client: { application: { name: "MongoDB Shell" }, driver: { name: "MongoDB Internal Client", version: "3.5.1-154-g166a966" }, os: { type: "Linux", name: "Amazon Linux AMI release 2016.03", architecture: "x86_64", version: "Kernel 4.4.15-25.57.amzn1.x86_64" } } } numYields:0 reslen:189 locks:{} protocol:op_query 0ms
2017-01-17T15:17:45.506+0000 I COMMAND  [conn14] command admin.$cmd appName: "MongoDB Shell" command: whatsmyuri { whatsmyuri: 1 } numYields:0 reslen:47 locks:{} protocol:op_command 0ms
2017-01-17T15:17:45.507+0000 I COMMAND  [conn14] command admin.$cmd appName: "MongoDB Shell" command: buildInfo { buildinfo: 1.0 } numYields:0 reslen:1265 locks:{} protocol:op_command 0ms
2017-01-17T15:17:45.507+0000 I STORAGE  [conn14] Reconfiguring WiredTiger storage engine with config string: "verbose=(dump_txn)"
2017-01-17T15:17:45.508+0000 I STORAGE  [conn14] WiredTiger message =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
2017-01-17T15:17:45.508+0000 I STORAGE  [conn14] WiredTiger message transaction state dump
2017-01-17T15:17:45.508+0000 I STORAGE  [conn14] WiredTiger message current ID: 40
2017-01-17T15:17:45.508+0000 I STORAGE  [conn14] WiredTiger message last running ID: 40
2017-01-17T15:17:45.508+0000 I STORAGE  [conn14] WiredTiger message oldest ID: 40
2017-01-17T15:17:45.508+0000 I STORAGE  [conn14] WiredTiger message oldest named snapshot ID: 0
2017-01-17T15:17:45.508+0000 I STORAGE  [conn14] WiredTiger message checkpoint running? no
2017-01-17T15:17:45.508+0000 I STORAGE  [conn14] WiredTiger message checkpoint generation: 2
2017-01-17T15:17:45.508+0000 I STORAGE  [conn14] WiredTiger message checkpoint pinned ID: 0
2017-01-17T15:17:45.508+0000 I STORAGE  [conn14] WiredTiger message checkpoint txn ID: 0
2017-01-17T15:17:45.508+0000 I STORAGE  [conn14] WiredTiger message session count: 16
2017-01-17T15:17:45.508+0000 I STORAGE  [conn14] WiredTiger message Transaction state of active sessions:
2017-01-17T15:17:45.508+0000 I STORAGE  [conn14] WiredTiger message =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
2017-01-17T15:17:45.508+0000 I COMMAND  [conn14] command admin.$cmd appName: "MongoDB Shell" command: setParameter { setParameter: 1.0, wiredTigerEngineRuntimeConfig: "verbose=(dump_txn)" } numYields:0 reslen:32 locks:{} protocol:op_command 0ms
2017-01-17T15:17:45.510+0000 I -        [conn14] end connection 127.0.0.1:45110 (1 connection now open)
2017-01-17T15:17:45.557+0000 I NETWORK  [thread1] connection accepted from 127.0.0.1:45112 #15 (1 connection now open)
2017-01-17T15:17:45.557+0000 I NETWORK  [conn15] received client metadata from 127.0.0.1:45112 conn15: { application: { name: "MongoDB Shell" }, driver: { name: "MongoDB Internal Client", version: "3.5.1-154-g166a966" }, os: { type: "Linux", name: "Amazon Linux AMI release 2016.03", architecture: "x86_64", version: "Kernel 4.4.15-25.57.amzn1.x86_64" } }
2017-01-17T15:17:45.557+0000 I COMMAND  [conn15] command admin.$cmd appName: "MongoDB Shell" command: isMaster { isMaster: 1, client: { application: { name: "MongoDB Shell" }, driver: { name: "MongoDB Internal Client", version: "3.5.1-154-g166a966" }, os: { type: "Linux", name: "Amazon Linux AMI release 2016.03", architecture: "x86_64", version: "Kernel 4.4.15-25.57.amzn1.x86_64" } } } numYields:0 reslen:189 locks:{} protocol:op_query 0ms
2017-01-17T15:17:45.557+0000 I COMMAND  [conn15] command admin.$cmd appName: "MongoDB Shell" command: whatsmyuri { whatsmyuri: 1 } numYields:0 reslen:47 locks:{} protocol:op_command 0ms
2017-01-17T15:17:45.558+0000 I COMMAND  [conn15] command admin.$cmd appName: "MongoDB Shell" command: buildInfo { buildinfo: 1.0 } numYields:0 reslen:1265 locks:{} protocol:op_command 0ms
2017-01-17T15:17:45.558+0000 I STORAGE  [conn15] Reconfiguring WiredTiger storage engine with config string: "verbose=(dump_cache)"
2017-01-17T15:17:45.558+0000 I STORAGE  [conn15] WiredTiger message =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
2017-01-17T15:17:45.558+0000 I STORAGE  [conn15] WiredTiger message cache dump
2017-01-17T15:17:45.558+0000 I STORAGE  [conn15] WiredTiger message file:index-6--7925628404692666237.wt(<live>):
2017-01-17T15:17:45.558+0000 I STORAGE  [conn15] WiredTiger message internal: 1 pages, 0MB, 1/0 clean/dirty pages, 0/0 clean/dirty MB, 0MB max page, 0MB max dirty page
2017-01-17T15:17:45.558+0000 I STORAGE  [conn15] WiredTiger message leaf: 1 pages, 0MB, 1/0 clean/dirty pages, 0/0 clean/dirty MB, 0MB max page, 0MB max dirty page
2017-01-17T15:17:45.558+0000 I STORAGE  [conn15] WiredTiger message file:collection-5--7925628404692666237.wt(<live>):
2017-01-17T15:17:45.558+0000 I STORAGE  [conn15] WiredTiger message internal: 1 pages, 0MB, 1/0 clean/dirty pages, 0/0 clean/dirty MB, 0MB max page, 0MB max dirty page
2017-01-17T15:17:45.558+0000 I STORAGE  [conn15] WiredTiger message leaf: 1 pages, 0MB, 1/0 clean/dirty pages, 0/0 clean/dirty MB, 0MB max page, 0MB max dirty page
2017-01-17T15:17:45.559+0000 I STORAGE  [conn15] WiredTiger message file:index-4--7925628404692666237.wt(<live>):
2017-01-17T15:17:45.559+0000 I STORAGE  [conn15] WiredTiger message internal: 1 pages, 0MB, 1/0 clean/dirty pages, 0/0 clean/dirty MB, 0MB max page, 0MB max dirty page
2017-01-17T15:17:45.559+0000 I STORAGE  [conn15] WiredTiger message leaf: 1 pages, 0MB, 1/0 clean/dirty pages, 0/0 clean/dirty MB, 0MB max page, 0MB max dirty page
2017-01-17T15:17:45.559+0000 I STORAGE  [conn15] WiredTiger message file:index-3--7925628404692666237.wt(<live>):
2017-01-17T15:17:45.559+0000 I STORAGE  [conn15] WiredTiger message internal: 1 pages, 0MB, 1/0 clean/dirty pages, 0/0 clean/dirty MB, 0MB max page, 0MB max dirty page
2017-01-17T15:17:45.559+0000 I STORAGE  [conn15] WiredTiger message leaf: 1 pages, 0MB, 1/0 clean/dirty pages, 0/0 clean/dirty MB, 0MB max page, 0MB max dirty page
2017-01-17T15:17:45.559+0000 I STORAGE  [conn15] WiredTiger message file:collection-2--7925628404692666237.wt(<live>):
2017-01-17T15:17:45.559+0000 I STORAGE  [conn15] WiredTiger message internal: 1 pages, 0MB, 1/0 clean/dirty pages, 0/0 clean/dirty MB, 0MB max page, 0MB max dirty page
2017-01-17T15:17:45.559+0000 I STORAGE  [conn15] WiredTiger message leaf: 1 pages, 0MB, 1/0 clean/dirty pages, 0/0 clean/dirty MB, 0MB max page, 0MB max dirty page
2017-01-17T15:17:45.559+0000 I STORAGE  [conn15] WiredTiger message file:index-1--7925628404692666237.wt(<live>):
2017-01-17T15:17:45.559+0000 I STORAGE  [conn15] WiredTiger message internal: 1 pages, 0MB, 1/0 clean/dirty pages, 0/0 clean/dirty MB, 0MB max page, 0MB max dirty page
2017-01-17T15:17:45.559+0000 I STORAGE  [conn15] WiredTiger message leaf: 1 pages, 0MB, 1/0 clean/dirty pages, 0/0 clean/dirty MB, 0MB max page, 0MB max dirty page
2017-01-17T15:17:45.559+0000 I STORAGE  [conn15] WiredTiger message file:collection-0--7925628404692666237.wt(<live>):
2017-01-17T15:17:45.559+0000 I STORAGE  [conn15] WiredTiger message internal: 1 pages, 0MB, 1/0 clean/dirty pages, 0/0 clean/dirty MB, 0MB max page, 0MB max dirty page
2017-01-17T15:17:45.559+0000 I STORAGE  [conn15] WiredTiger message leaf: 1 pages, 0MB, 1/0 clean/dirty pages, 0/0 clean/dirty MB, 0MB max page, 0MB max dirty page
2017-01-17T15:17:45.559+0000 I STORAGE  [conn15] WiredTiger message file:_mdb_catalog.wt(<live>):
2017-01-17T15:17:45.559+0000 I STORAGE  [conn15] WiredTiger message internal: 1 pages, 0MB, 1/0 clean/dirty pages, 0/0 clean/dirty MB, 0MB max page, 0MB max dirty page
2017-01-17T15:17:45.559+0000 I STORAGE  [conn15] WiredTiger message leaf: 1 pages, 0MB, 1/0 clean/dirty pages, 0/0 clean/dirty MB, 0MB max page, 0MB max dirty page
2017-01-17T15:17:45.559+0000 I STORAGE  [conn15] WiredTiger message file:sizeStorer.wt(<live>):
2017-01-17T15:17:45.559+0000 I STORAGE  [conn15] WiredTiger message internal: 1 pages, 0MB, 1/0 clean/dirty pages, 0/0 clean/dirty MB, 0MB max page, 0MB max dirty page
2017-01-17T15:17:45.559+0000 I STORAGE  [conn15] WiredTiger message leaf: 1 pages, 0MB, 1/0 clean/dirty pages, 0/0 clean/dirty MB, 0MB max page, 0MB max dirty page
2017-01-17T15:17:45.559+0000 I STORAGE  [conn15] WiredTiger message file:WiredTigerLAS.wt(<live>):
2017-01-17T15:17:45.559+0000 I STORAGE  [conn15] WiredTiger message internal: 1 pages, 0MB, 1/0 clean/dirty pages, 0/0 clean/dirty MB, 0MB max page, 0MB max dirty page
2017-01-17T15:17:45.559+0000 I STORAGE  [conn15] WiredTiger message file:WiredTiger.wt(<live>):
2017-01-17T15:17:45.559+0000 I STORAGE  [conn15] WiredTiger message internal: 1 pages, 0MB, 1/0 clean/dirty pages, 0/0 clean/dirty MB, 0MB max page, 0MB max dirty page
2017-01-17T15:17:45.559+0000 I STORAGE  [conn15] WiredTiger message leaf: 1 pages, 0MB, 1/0 clean/dirty pages, 0/0 clean/dirty MB, 0MB max page, 0MB max dirty page
2017-01-17T15:17:45.559+0000 I STORAGE  [conn15] WiredTiger message cache dump: total found: 0MB vs tracked inuse 0MB
2017-01-17T15:17:45.559+0000 I STORAGE  [conn15] WiredTiger message total dirty bytes: 0MB
2017-01-17T15:17:45.559+0000 I STORAGE  [conn15] WiredTiger message =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
2017-01-17T15:17:45.559+0000 I COMMAND  [conn15] command admin.$cmd appName: "MongoDB Shell" command: setParameter { setParameter: 1.0, wiredTigerEngineRuntimeConfig: "verbose=(dump_cache)" } numYields:0 reslen:32 locks:{} protocol:op_command 1ms
2017-01-17T15:17:45.560+0000 I -        [conn15] end connection 127.0.0.1:45112 (1 connection now open)
```

File: src/conn/conn_api.c [hunk 1]
```diff
@@ -1149,6 +1150,13 @@ __conn_reconfigure(WT_CONNECTION *wt_conn, const char *config)
 	__wt_free(session, conn->cfg);
 	conn->cfg = p;
 
+	/*
+	 * Fourth, run any diagnostic commands that were included. Do that last,
+	 * reconfiguration might include other configuration changes to support
+	 * a command.
+	 */
+	WT_ERR(__conn_verbose_cmd(session, config));
```
Comment (agorrod) [inline src/conn/conn_api.c]: Would it make sense to move this into {{__wt_verbose_config}} and short-circuit there if verbose is the only option specified. I guess that requires doing some parsing of the config string, but potentially restarting all the utility threads each time an application tries to output a verbose message makes me nervous. Phrased another way: Until this point I've been thinking of reconfigure as a heavy weight operation that is likely to trigger significant changes to the running environment. This functionality is more like "dump some info" and I'd prefer if they weren't entangled.

I feel like I'm selling myself on adding a new API instead of using ``WT_CONNECTION::reconfigure``, but that has back-porting consequences and means more plumbing will need to be added to MongoDB to expose the functionality.

File: src/conn/conn_api.c [hunk 2]
```diff
@@ -9,6 +9,7 @@
 #include "wt_internal.h"
 
 static int __conn_statistics_config(WT_SESSION_IMPL *, const char *[]);
+static int __conn_verbose_cmd(WT_SESSION_IMPL *, const char *);
```
Comment (agorrod) [inline src/conn/conn_api.c]: There has been another case recently where verbose has been used to expose more information to MongoDB, which is ``recovery_progress`` - that is different to these options, since it's a general flag setting rather than a single request for information.

Comment (keithbostic) [inline src/conn/conn_api.c]: > I've got some concerns about how `WT_CONNECTION::reconfigure()` with `verbose` is being used and seems to be being expanded moving forward.

I prefer:
```
WT_CONNECTION.reconfigure("verbose=(dump_txn,dump_cache)"))
```
to the previous:
```
WT_CONNECTION.reconfigure("diagnostic=(dump_txn,dump_cache)")
```
having this functionality in "verbose" makes sense to me. That said, I don't have a strong opinion, I'm happy to go with whatever you and @michaelcahill decide.

File: src/conn/conn_api.c (hunk 1 above)
Comment (keithbostic) [inline src/conn/conn_api.c]: > Would it make sense to move this into `__wt_verbose_config`

I thought about that myself, and didn't do it for two reasons: first, `__wt_verbose_config` doesn't take the `config` argument, and second, it means `__wt_verbose_config` has to be the last configuration function we call (so it's possible to configure stuff so a subsequent "diagnostic dump" command will work), and that seemed like a way to introduce a bug some day in the future. I don't feel strongly, let me know if you still prefer that change.

> and short-circuit there if verbose is the only option specified.

These APIs aren't likely to be called frequently: I think admins are more likely to commonly reconfigure more eviction worker threads, for example.

I think @michaelcahill went this way because "verbose" isn't an unreasonable name for this functionality, and implies MongoDB requires no additional plumbing.

Comment (sueloverso) [inline src/conn/conn_api.c]: > These APIs aren't likely to be called frequently: I think admins are more likely to commonly reconfigure more eviction worker threads, for example.

I generally do not agree with this statement.  I doubt any admins or mongodb support people have reconfigured eviction worker threads much other than an explicit direction to do so from someone on our team in some HELP ticket.  Given the number of HELP tickets that discuss eviction, and once this cat is out of the bag, I can see this being used more and more frequently.

Basically I agree with @agorrod that short-circuiting and avoiding thread disruption on verbose-only is a good idea.  I think it will be used a lot more often than you're expecting.

Comment (keithbostic) [inline src/conn/conn_api.c]: > and avoiding thread disruption on verbose-only is a good idea.

To be clear, there shouldn't be any thread disruption, thread disruption would be evidence of a bug in the code.  That said...

Comment (sueloverso) [inline src/conn/conn_api.c]: Thanks Keith.  I should have done a code perusal first.  You are correct that if nothing is changing we do not stop/restart threads along the way.

File: src/conn/conn_api.c [hunk 3]
```diff
@@ -1855,6 +1863,45 @@ __wt_verbose_config(WT_SESSION_IMPL *session, const char *cfg[])
...
+ */
+static int
+__conn_verbose_cmd(WT_SESSION_IMPL *session, const char *config)
+{
+	static const struct verbose_cmd_struct {
+		const char *name;			/* command name */
+		int (*func)(WT_SESSION_IMPL *);		/* command */
+	} verbose_cmd[] = {
+		 { "dump_cache", __wt_verbose_dump_cache },
+		 {   "dump_txn", __wt_verbose_dump_txn },
```
Comment (sueloverso) [inline src/conn/conn_api.c]: Some whitespace issue going on there.  I think this function needs a different name and/or a big comment.  Maybe call it `conn_verbose_once_cmd` to make it obvious it is only for one-time diagnostic.  Or `conn_verbose_diagnostics`.

File: src/txn/txn.c [hunk 4]
```diff
@@ -803,3 +803,95 @@ __wt_txn_global_destroy(WT_SESSION_IMPL *session)
...
+
+	WT_RET(__wt_msg(session, "checkpoint running? %s",
+	    txn_global->checkpoint_running ? "yes" : "no"));
+	WT_RET(__wt_msg(session,
+	    "checkpoint generation: %" PRIu64, txn_global->checkpoint_gen));
+	WT_RET(__wt_msg(session,
+	    "checkpoint pinned ID: %" PRIu64, txn_global->checkpoint_pinned));
+	WT_RET(__wt_msg(session,
+	    "checkpoint txn ID: %" PRIu64, txn_global->checkpoint_txnid));
+	WT_RET(__wt_msg(session, "session count: %" PRIu32, session_cnt));
```
Comment (sueloverso) [inline src/txn/txn.c]: I suggest moving the ORDERED_READ down to here just before we need it.  I think message calls could be slow such that the value could be stale by the time we get here.  It still could be stale, but maybe less so.

File: src/conn/conn_api.c (hunk 3 above)
Comment (keithbostic) [inline src/conn/conn_api.c]: Sorry, trying to be too clever and making the columns line up. I've pushed a cleanup.

File: src/txn/txn.c (hunk 4 above)
Comment (keithbostic) [inline src/txn/txn.c]: Sure, fixed.

**Comment by michaelcahill** [comment]
@keithbostic, I'm just catching up on this -- it's not quite what I expected: I thought we were going away from a special command and making this a genuine verbose mode.

Above, you wrote "I'm dumping the transaction/cache information every 5 minutes, as long as we're stuck" -- is that still valid, and if so, where is it in the code?  I ask because that is what I expected: if eviction is stuck and this verbose mode was configured, do the dump at that point.

**Comment by keithbostic** [comment]
@michaelcahill, @agorrod 

>  I'm just catching up on this -- it's not quite what I expected: I thought we were going away from a special command and making this a genuine verbose mode.

I must have missed that part: my understanding from talking to you & Bruce was you wanted the ability to explicitly dump the state of the system using this command. That seems generally useful to me, to dump the state of the transactional subsystem when eviction isn't stuck, for example.

> Above, you wrote "I'm dumping the transaction/cache information every 5 minutes, as long as we're stuck" -- is that still valid, and if so, where is it in the code? I ask because that is what I expected: if eviction is stuck and this verbose mode was configured, do the dump at that point.

Look at the changes to `__evict_server()`: the DIAGNOSTIC mode checks have changed, and regardless of any verbose setting we're dumping the state of the cache and transaction subsystems every 5 minutes, if eviction is stuck.

Just let me know what changes you want and I'll clean it up.

**Comment by michaelcahill** [comment]
@keithbostic, what I'd like is for there to be a new verbose mode that controls whether or not the eviction server dumps the cache and transaction state when it gets stuck.  I could imagine MongoDB configuring it on by default, and would expect to do the same in automated testing (probably via the global `WIREDTIGER_CONFIG` setting).

I'm ambivalent about whether to have a single `evict_stuck` mode or separate `evict_stuck_cache` and `evict_stuck_transaction` flags -- I suspect folks would either want neither or both.

I was never comfortable overloading `reconfigure` to perform one-off diagnostic commands (even though we talked about this early on).  I think an ordinary verbose flag would suffice for the common use cases I can think of.

**Comment by keithbostic** [comment]
> I'd like is for there to be a new verbose mode that controls whether or not the eviction server dumps the cache and transaction state when it gets stuck. I could imagine MongoDB configuring it on by default, and would expect to do the same in automated testing (probably via the global `WIREDTIGER_CONFIG` setting).

OK, I've pushed something pretty close to that.

I left the `HAVE_DIAGNOSTIC` code that dumps the cache/transaction state in place; we need the `#ifdef HAVE_DIAGNOSTIC` (assuming we still want to exit in that case), so there's no advantage to removing the dump of the cache/transaction state, and it means we don't have to separately configure `WIREDTIGER_CONFIG` for automated testing.

You didn't say how MongoDB would configure this new verbose option (since MongoDB doesn't build with `HAVE_VERBOSE`). Commit 6ced66c makes the `evict_stuck` verbose configuration flag independent of `HAVE_VERBOSE` being configured in the build. If that's not the way you wanted it done, it's a separate commit so it can be simply reverted.

**Comment by keithbostic** [comment]
autobuild: retest this please

**Comment by michaelcahill** [comment]
Thanks, @keithbostic -- this lgtm now.  I'll merge.
